File: abc/abc115/abc115_b.cs
// https://atcoder.jp/contests/abc115/submissions/16817334

using System;
using System.Linq;

class Program
{
    static void Main()
    {
        var N = int.Parse(Console.ReadLine());
        var p = new int[N];
        for (int i = 0; i < N; i++)
        {
            p[i] = int.Parse(Console.ReadLine());
        }
        Console.WriteLine(p.Sum() - p.Max() / 2);
    }
}

File: abc/abc115/abc115_b.fs
// https://atcoder.jp/contests/abc115/submissions/16817863

let N = stdin.ReadLine() |> int

[ for i in [ 0..N - 1 ] do
    yield (stdin.ReadLine() |> int) ]
|> fun x -> (List.sum x - List.max x / 2)
|> stdout.WriteLine
